exercise.wwwapi/Endpoint/ProductEndpoint.cs

```diff
@@ -0,0 +1,86 @@
+﻿using exercise.wwwapi.Data;
+using exercise.wwwapi.Models;
+using exercise.wwwapi.Repository;
+using Microsoft.AspNetCore.DataProtection.Repositories;
+using Microsoft.AspNetCore.Mvc;
+
+namespace exercise.wwwapi.Endpoint
+{
+    public static class ProductEndpoint
+    {
+        public static void ConfigureProductEndpoint(this WebApplication app)
+        {
+            var productGroup = app.MapGroup("products");
+
+
+            productGroup.MapGet("", GetProducts);
+            productGroup.MapGet("/{id}", GetProduct);
+            productGroup.MapPost("/{id}", AddProduct);
```

Comment:
Id is not required. Since you don`t know that Id you are getting cause you are creating a new entity